lib/cequel/metal/keyspace.rb

```diff
@@ -316,7 +316,12 @@ def extract_hosts_and_port(configuration)
       end
 
       def extract_credentials(configuration)
-        configuration.slice(:username, :password).presence
+        credentials = configuration.slice(:username, :password).presence
+        
+        return credentials unless configuration[:auth_provider]
+
+        { auth_provider: CequelPassword.new(*credentials.values),
+          datacenter: configuration[:datacenter].presence }
```

Comment:
This seems to ignore the value of `configuration[:auth_provider]` which would surprise me, as a user.

Also, if we start supporting auth providers perhaps we should just go whole hog and always use an auth provider, even when `:username` and `:password` are specified in the configuration. Maybe something like

```
def extract_auth_provider
  auth_provider = configuration.fetch(:auth_provider, Cequel:Auth:Provider:Password)
  auth_provider.new(configuration)
end
```
```
class Cequel:Auth:Provider:Password
  def initialize(config)
    @user = config.fetch(:username)
    @pass = config.fetch(:password)
  end
  def create_authenticator(_)
    Cassandra::Auth::Providers::Password::Authenticator.new(@user, @pass)
  end
end